File: Homechef_Group-19_e-DAC May 2021/homechefDemo_2021-09-29_CompleteBackend/homechefDemo/src/main/java/com/app/springboot/homechefDemo/service/SpecialityService.java
package com.app.springboot.homechefDemo.service;

import java.io.IOException;
import java.util.List;

import org.springframework.web.multipart.MultipartFile;

import com.app.springboot.homechefDemo.entity.Chef;
import com.app.springboot.homechefDemo.entity.Speciality;

public interface SpecialityService {
	
	public List<Speciality> findAll();
	public int getSpecialityCount();
	public Chef findById(int cuisineid); 
	public void save(Chef theSpeciality);
	public void deleteById(int cuisineid);
	void updateSpeciality(int cuisineid, Speciality sp); 
	public List<Speciality> getSpecialitybychefid(int chefid);
	public void saveFile(int cuisineid, String fileName, MultipartFile multipartFile) throws IOException;
	public void updateSpecialityImage(int cuisineid, String fileName);
	public String findSpecialityImageById(int cuisineid); 
	
	
}
	
	

File: Homechef_Group-19_e-DAC May 2021/homechef_frontend/src/services/SpecialityService.js
import axios from 'axios';

const CHEF_API_BASE_URL = "http://localhost:8080/homechefDemo";
//const CHEF_API_SPECIALITY_DEATILS_BY_CHEF_ID_URL= CHEF_API_BASE_URL +'/'+'speciality' + 'getSpecialitybychefid/'


class SpecialityService{

    viewSpeciality(){
        return axios.get(CHEF_API_BASE_URL +'/speciality/allSpeciality');
    }

   /*  getSpecialityDetailsById(chefid){
        return axios.get(CHEF_API_SPECIALITY_DEATILS_BY_CHEF_ID_URL + cuisineid);
    } */

}


export default new SpecialityService()